framework/branch-key-store.md

```diff
@@ -431,74 +625,128 @@ If the Keystore's KMS Configuration is `Discovery` or `MRDiscovery`,
+      MUST be added with an Attribute Name of `aws-crypto-ec:` + the Key and Attribute Value (S) of the value.
+  - ConditionExpression: `attribute_exists(branch-key-id) AND enc = :encOld`
+  - ExpressionAttributeValues: `{":encOld" := DDB.AttributeValue.B(oldCiphertextBlob)}`
+  - TableName: the configured Table Name
+
+TransactWriteItemRequest:
+
+- TransactWriteItems: List of TransactWriteItem
+
+If DDB TransactWriteItems is successful, this operation MUST return a successful response containing no additional data.
 Otherwise, this operation MUST yield an error.
 
-#### Authenticating an EncryptedHierarchicalKey
+The condition expression for the Active Input ensures
+the Active Item in storage has not changed since it was read.
+This prevents overwrites due to a race in updating the Active Item.
+
+#### Authenticating a Keystore item for item with `hierarchy-version` v1
 
 The operation MUST use the configured `KMS SDK Client` to authenticate the value of the keystore item.
 
+Every attribute on the AWS DDB response item will be authenticated.
+
+Every key in the constructed [encryption context](#encryption-context)
+except `tableName`
+MUST exist as a string attribute in the AWS DDB response item.
+Every value in the constructed [encryption context](#encryption-context)
+except the logical table name
+MUST equal the value with the same key in the AWS DDB response item.
+The key `enc` MUST NOT exist in the constructed [encryption context](#encryption-context).
+
 The operation MUST call [AWS KMS API ReEncrypt](https://docs.aws.amazon.com/kms/latest/APIReference/API_ReEncrypt.html)
 with a request constructed as follows:
 
-- `SourceEncryptionContext` MUST be the [encryption context](#encryption-context) of the EncryptedHierarchicalKey to be authenticated
+- `SourceEncryptionContext` MUST be the [encryption context](#encryption-context) constructed above
 - `SourceKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured KMS Key in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
-- `CiphertextBlob` MUST be the `CiphertextBlob` attribute value on the EncryptedHierarchicalKey to be authenticated
+- `CiphertextBlob` MUST be the `enc` attribute value on the AWS DDB response item
 - `GrantTokens` MUST be the configured [grant tokens](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#grant_token).
 - `DestinationKeyId` MUST be [compatible with](#aws-key-arn-compatibility) the configured KMS Key in the [AWS KMS Configuration](#aws-kms-configuration) for this keystore.
-- `DestinationEncryptionContext` MUST be the [encryption context](#encryption-context) of the EncryptedHierarchicalKey to be authenticated
+- `DestinationEncryptionContext` MUST be the [encryption context](#encryption-context) constructed above
+
+#### Authenticating a Keystore item for item with `hierarchy-version` v2
```

Comment:
Nit: Branch Key Store item.

There are many Key Stores in the world. 
Ours is the Branch Key Store.

Reply:
Update it as `Authenticating a Branch Keystore item for item with hierarchy-version v(1/2)`